src/System.Text.Encoding/tests/UTF32Encoding/UTF32EncodingDecode.cs

```diff
@@ -102,27 +102,30 @@ public void Decode_InvalidBytes()
             Decode_InvalidBytes(new byte[] { 0xFF, 0xFF, 0xFF, 0xFF }, 0, 4, "\uFFFD");
         }
 
-        public static byte[] GetBigEndianBytes(byte[] littleEndianBytes)
+        public static byte[] GetBigEndianBytes(byte[] littleEndianBytes, int index, int count)
```

Comment:
maybe you can try implementation like the following:

```
        public static byte[] GetBigEndianBytes(byte[] littleEndianBytes, int index, int count)
        { 
            byte[] bytes = new byte[count];

            int i;
            for (i = 0; i+3<count; i += 4)
            { 
                bytes[i] = littleEndianBytes[i + index + 3]; 
                bytes[i + 1] = littleEndianBytes[i + index + 2];
                bytes[i + 2] = littleEndianBytes[i + index + 1];
                bytes[i + 3] = littleEndianBytes[i + index];
            }

            // even I prefer we always pass 4-bytes aligned array and assert if we get more bytes here
            // this will be more reliable as the last left bytes will not be in right Endian forms
            for (; i<count; i++)
            {
                bytes[i] = littleEndianBytes[i + index];
            }

            return bytes; 
        } 

```